pkg/storage/utils/decomposedfs/spaces.go

```diff
@@ -1008,7 +1018,28 @@ func (fs *Decomposedfs) storageSpaceFromNode(ctx context.Context, n *node.Node,
 
 	// add rootinfo
 	ps, _ := n.SpaceRoot.PermissionSet(ctx)
-	space.RootInfo, _ = n.SpaceRoot.AsResourceInfo(ctx, &ps, nil, nil, false)
+	space.RootInfo, _ = n.SpaceRoot.AsResourceInfo(ctx, &ps, []string{"quota"}, nil, false)
+
+	// we cannot put free, used and remaining into the quota, as quota, when set would always imply a quota limit
+	// for now we use opaque properties with a 'quota.' prefix
+	quotaStr := node.QuotaUnknown
+	if space.RootInfo.Opaque != nil && space.RootInfo.Opaque.Map != nil && space.RootInfo.Opaque.Map["quota"] != nil && space.RootInfo.Opaque.Map["quota"].Decoder == "plain" {
+		quotaStr = string(space.RootInfo.Opaque.Map["quota"].Value)
+	}
```

Comment:
How about changing this to

```suggestion

	if quotaInOpaque := sdk.DecodeOpaqueMap(space.RootInfo.Opaque)["quota"]; quotaInOpaque != "" {
		quotaStr = quotaInOpaque
	}
```
This also requires a `sdk "github.com/cs3org/reva/v2/pkg/sdk/common"` import, though.